pep-0517.txt

```diff
@@ -166,48 +166,71 @@ The build backend object is expected to have attributes which provide
 some or all of the following hooks. The common ``config_settings``
 argument is described after the individual hooks::
 
-  def get_build_requires(config_settings):
+  def build_sdist(source_directory, sdist_directory, config_settings):
+      ...
+
+MUST build an unpacked sdist from the ``source_directory`` and place it into the
+specified ``sdist_directory``. If the ``source_directory`` is already an
```

Comment:
The discrepancy is because as of today, there is no real standard for metadata inside of a sdist. The only real standard is "a tarball with a ``setup.py`` script that looks close to what setuptools supports". This PEP extends that to also include "a tarball with a ``pyproject.toml`` file with a particular key". We currently call ``python setup.py egg_info`` to get the metadata that "belongs" to the sdist (it doesn't really, it belongs to the emphereal binary output that the sdist creates), and with this PEP we can just call the ``get_wheel_metadata`` hook to achieve the same thing.

If we create a sdist 2.0 then we can focus on what metadata we want to include in an sdist to enable some other things that would be nice to have, but like you said that opens up a huge debate. This PEP is "let me replace ``python setup.py bdist_wheel`` with something else", and my statement is that we also need to replace ``python setup.py sdist``, not that we need to drastically alter the status quo.

And yes, they'll have to track any work for a sdist 2.0 (or well they don't have to, just if they don't they'll never gain any improvements we get from that), just like they'll have to track a hypothetical wheel 2.0.